typeutil.go

```diff
@@ -165,20 +166,32 @@ func newFileScope(imp types.Importer, pkg *types.Package) *fileScope {
 	return &fileScope{otherNames: make(map[string]bool), pkg: pkg, imp: imp}
 }
 
+func unvendor(path string) string {
+	// path might include "vendor"
+	// These paths do not compile, so we need to remove everything
+	// up to and including "/vendor/"
+	// see https://github.com/golang/go/issues/12019
+	if i := strings.LastIndex(path, "/vendor/"); i != -1 {
+		path = path[i+len("/vendor/"):]
+	}
+	return path
+}
+
 func (s *fileScope) writeImportDecl(w io.Writer) {
 	fmt.Fprintln(w, "import (")
 	for _, pkg := range s.imports {
 		if s.importNames[pkg.Path()] != pkg.Name() {
 			fmt.Fprintf(w, "\t%s %q\n", s.importNames[pkg.Path()], pkg.Path())
```

Comment:
This needs to be `unvendor`-ed too:

```suggestion
			fmt.Fprintf(w, "\t%s %q\n", s.importNames[pkg.Path()], unvendor(pkg.Path()))
```

BTW, thanks for this fix!